app/src/main/java/fr/neamar/kiss/SettingsActivity.java

```diff
@@ -36,9 +37,15 @@ public void onResume() {
         prefs.registerOnSharedPreferenceChangeListener(this);
     }
 
+    @SuppressWarnings("deprecation")
     @Override
     public void onSharedPreferenceChanged(SharedPreferences sharedPreferences, String key) {
 
+        if (key.equals("history-hide")) {
+            findPreference("history-onclick").setEnabled(prefs.getBoolean(key, true));
```

Comment:
for this kind of beahvior, better to instead use `android:dependency=` in the XML to enable / disable the preference automatically